src/Bedrock.Framework.Experimental/Protocols/Http1HeaderReader.cs

```diff
@@ -0,0 +1,91 @@
+﻿using Bedrock.Framework.Infrastructure;
+using System;
+using System.Buffers;
+using System.Text;
+
+namespace Bedrock.Framework.Protocols.Http.Http1
+{
+    public class Http1HeaderReader : IMessageReader<ParseResult<Http1Header>>
+    {
+        private const byte SP = (byte)' ';
+        private const byte HT = (byte)'\t';
+        private const byte COLON = (byte)':';
+        private const byte CR = (byte)'\r';
+        private const byte LF = (byte)'\n';
+
+        private static readonly byte[] _colonOrWhitespace = new[] { COLON, CR, SP, HT };
```

Comment:
The pattern should look like this: https://github.com/davidfowl/BedrockFramework/blob/5b7d03dc94edeb202afbd0a813ab41fa8a91612d/src/Bedrock.Framework.Experimental/Protocols/ChunkedHttpBodyReader.cs#L11

Naming and should be using `ReadOnlySpan<byte>`, you can read more about the compiler feature here https://vcsjones.dev/2019/02/01/csharp-readonly-span-bytes-static

Reply:
I'm not sure I agree with calling it new line - the way its used is looking for `CR` or `LF`, not `CR` followed by an `LF`.

Also I'm not sure why we don't reuse the constants defined above rather than repeating them?

Reply:
I'm not sure what you're asking. What's not clear?

Reply:
You suggested using `NewLine` for the name, and in your example you replaced `CR` with `(byte)'\r'`.

I was saying that the name didn't seem correct, and asking why you did the latter.

Reply:
Oh I wasn't talking about the naming, I meant the style (no underscore because its a property) and using the span trick.